src/Build/BuildCheck/Infrastructure/BuildCheckConnectorLogger.cs

```diff
@@ -75,17 +90,69 @@ private void EventSource_AnyEventRaised(object sender, BuildEventArgs e)
 
     private void EventSource_BuildFinished(object sender, BuildFinishedEventArgs e)
     {
-        _stats.Merge(buildCheckManager.CreateTracingStats(), (span1, span2) => span1 + span2);
-        string msg = string.Join(Environment.NewLine, _stats.Select(a => a.Key + ": " + a.Value));
-
         BuildEventContext buildEventContext = e.BuildEventContext ?? new BuildEventContext(
             BuildEventContext.InvalidNodeId, BuildEventContext.InvalidTargetId,
             BuildEventContext.InvalidProjectContextId, BuildEventContext.InvalidTaskId);
 
         LoggingContext loggingContext = loggingContextFactory.CreateLoggingContext(buildEventContext);
 
-        // Tracing: https://github.com/dotnet/msbuild/issues/9629
-        loggingContext.LogCommentFromText(MessageImportance.High, msg);
+        _stats.Merge(buildCheckManager.CreateAnalyzerTracingStats()!, (span1, span2) => span1 + span2);
+        LogAnalyzerStats(loggingContext);
+    }
+    
+    private void LogAnalyzerStats(LoggingContext loggingContext)
```

Comment:
The `LogCommentFromText` will write to binlog if attached as well - however that's an unstructured representation, so would need to be searched and parsed with proprietary and fragile code. Log a structured data with `loggingContext.LogBuildEvent` as well.

You do not need to create new BuildEvent for that purpose - the existing `BuildCheckTracingEventArgs` can be reused for this purpose. Just add a boolean flag which indicates whether it's a single node intermediary data, or final merged stats (so that binlog viewer code can decide which should be ignored and which should be parsed)